docs/src/pages/testing/playwright.mdx

```diff
@@ -0,0 +1,82 @@
+import { Callout } from "nextra-theme-docs";
+
+# Testing with Playwright
+
+Playwright is a end-to-end testing tool that allows you to test your web
+applications in a browser.
+
+We can use Playwright to create **integration tests** for our `UploadThing`
+funcionality, independently of the frontend framework we are using.
+
+<Callout type="info">
+  Integration tests are used to test the interaction between different parts of
+  your application. In the case of `UploadThing`, we can use Playwright to test
+  the interaction between the frontend and the backend.
+</Callout>
+
+## Installation
+
+If you haven't already, install Playwright by running:
+
+```bash
+npm install -D playwright
+```
+
+## Writing the test
+
+When writing your test, you can think of it as a user interacting with your
+application. For example, a real user would:
+
+1. Open the browser
+2. Go to the page where the file upload form is
+3. Upload a file
+4. Click the submit button
+5. Check if the file was uploaded successfully
+
+But instead of uploading a real file to `Uploadthing` cloud storage, we will
+mock the API endpoint and check if the frontend is sending the file correctly.
+
+```ts
+import { expect, test } from "@playwright/test";
+
+test("should upload a file successfully", async ({ page }) => {
+  // Mock the API endpoint, so we don't actually upload a file
+  // This is a Next.js API route, but you can replace it with your own
+  await page.route(
```

Comment:
This feels like the wrong place to be mocking, since the api code is also provided by uploadthing. 

might make more sense to be mocking the API calls to uploadthing.com/api that the server code does, but even still I am having difficulty understanding what you are trying to do here.

Reply:
Makes sense to me, specially if the example is suppose to be FE framework agnostic

Reply:
You may want to take a look at how we are doing our mocks with MSW for some inspiration there. 

https://github.com/pingdotgg/uploadthing/blob/3376306731d34c36fea8df3c6b117eeb17328f0f/packages/uploadthing/test/__test-helpers.ts

Reply:
Hey there, 

It result that it's pretty complicated to mock the uploadthing endpoints because of the nature of Next.js. Explanation: 

1. Next.js is fullstack framework.
2. Playwright is a E2E testing tool.
3. Playwright can only mocks the network requests that are made in the client side.
4. The Uploadthing library for Next.js approach is to create a Next.js api routes where internally is the logic and rules in order to actually fetch de files to the Uploadthing APIs. 
5. The Uploadthing internal implementation is executed in the server side, not exposed to the client.

Therefore, I was not able to mock the real Uploadthing endpoints 😞 

So probably having the example mocking the Next.js API route is the only way, so far. 

What I have seen is a [combination of Mock Service Worker with Playwright,](https://stackoverflow.com/questions/70980370/how-do-i-mock-server-side-api-calls-in-a-nextjs-app) but it looks more complex than I would like to see when working with testing.

What are your thoughts?